tests/ConfigControllerTest.php

```diff
@@ -19,31 +34,137 @@ public function testConfigurationIsSavedWithEmptyValues()
         $configController = new ConfigController();
         $configController->handleConfiguration($request);
 
-        $store = Store::where('domain', 'stores/test123')->first();
+        $store = Store::where('context', 'stores/test123')->first();
         $config = Config::where('store_id', $store['id'])->first();
 
         $this->assertFalse(boolval($config['active']));
         $this->assertEquals($store['id'], (int) $config['store_id']);
         $this->assertEmpty($config['shopkey']);
     }
 
-    public function testConfigurationIsSavedWithValues()
+    public function testHandleConfigurationSavesConfigAndScripts()
     {
-        $request = Request::create('/auth/load', 'POST', [
+        $mockedStoresXmlResponse = $this->getMockResponse('/xml_response/v2_stores.xml');
+        $mockedScriptsJsonResponse = $this->getMockResponse('json_response/v3_content_scripts.json');
+
+        $mock = new MockHandler([
+            // REST-API call: stores/<store_hash>/v2/store
+            new Response(200, ['Content-Type' => 'application/xml'], $mockedStoresXmlResponse),
+            // REST-API call: stores/<store_hash>/v3/content/scripts
+            new Response(200, ['Content-Type' => 'application/json'], $mockedScriptsJsonResponse),
+        ]);
+        $handler = HandlerStack::create($mock);
+        $httpClientMock = new Client(['handler' => $handler]);
+        $configController = new ConfigController($httpClientMock);
+
+        $request = Request::create('/config', 'POST', [
             'store_hash' => 'test123',
             'access_token' => 'testAcessToken123',
             'context' => 'stores/test123',
             'shopkey' => '123test',
             'active_status' => 'on'
         ]);
-        $configController = new ConfigController();
         $configController->handleConfiguration($request);
 
-        $store = Store::where('domain', 'stores/test123')->first();
+        /** @var Store $store */
+        $store = Store::where('context', 'stores/test123')->first();
         $config = Config::where('store_id', $store['id'])->first();
 
         $this->assertTrue(boolval($config['active']));
         $this->assertEquals($store['id'], (int) $config['store_id']);
         $this->assertSame('123test', $config['shopkey']);
+
+        $this->assertEquals(1, $store->scripts->count());
+        $script = $store->scripts()->first();
+
+        $this->assertEquals('Findologic', $script->name);
+        $this->assertEquals('9dd04fae-d45d-5a64-a45d-62d14d2c62b5', $script->uuid);
+        $this->assertEquals(str_replace('stores/', '', $store->context), $script->store_hash);
```

Comment:
In such cases I prefer `assertStringContainsString`. The simple reason for this is that I do not want to manipulate the outcoming data, because my manipulation could change the string in a way that makes this assertion pass, even when it shouldn't. This is more a preference than an actual issue.